Framework/Algorithms/test/CorelliPowderCalibrationDatabaseTest.h

```diff
@@ -0,0 +1,431 @@
+    CorelliCalibration::ComponentPosition goldsourcepos{0., 0., -15.560, 0.,
+                                                        0., 0., 0.};
+    CorelliCalibration::ComponentPosition testsourcepos =
+        calib_handler.getComponentCalibratedPosition("source");
+    TS_ASSERT(testsourcepos.equalTo(goldsourcepos, 1.E-10));
+
+    CorelliCalibration::ComponentPosition goldbank1pos{
+        0.9678, 0.0056, 0.0003, 0.4563, -0.9999, 0.3424, 5.67};
+    CorelliCalibration::ComponentPosition testbank1pos =
+        calib_handler.getComponentCalibratedPosition("bank1");
+    TS_ASSERT(testbank1pos.equalTo(goldbank1pos, 1.E-10));
+
+    // Test: save calibration table
+    // component file: name, remove file if it does exist, save and check file
+    // existence
+    const std::string testcalibtablefilename{"testsourcedb2.csv"};
+    boost::filesystem::remove(testcalibtablefilename);
+    calib_handler.saveCalibrationTable(testcalibtablefilename);
+    TS_ASSERT(boost::filesystem::exists(testcalibtablefilename));
+    // load file and check
+    TableWorkspace_sptr duptable =
+        loadCSVtoTable(testcalibtablefilename, "DuplicatedSource");
+    TS_ASSERT_EQUALS(duptable->rowCount(), 3);
+    TS_ASSERT_DELTA(duptable->cell<double>(2, 6), 0.3424, 0.00001);
+
+    // Test: load calibration table
+    // TODO - later. not important now
+    //    calib_handler.load(testcompfilename);
+    //    TableWorkspace_sptr compcalibws =
+    //    calib_handler.getCalibrationWorkspace();
+
+    // Test: save single component file
+    const std::string testsamplecalfilename{"/tmp/testsampledb2.csv"};
+    boost::filesystem::remove(testsamplecalfilename);
+    // save
+    calib_handler.saveCompomentDatabase("20201117", "sample",
+                                        testsamplecalfilename);
+    TS_ASSERT(boost::filesystem::exists(testsamplecalfilename));
+
+    // load
+    TableWorkspace_sptr dupsampletable =
+        calib_handler.loadComponentCalibrationTable(testsamplecalfilename,
+                                                    "TestSampleCalib1");
+    // check row number and value
+    TS_ASSERT_EQUALS(dupsampletable->rowCount(), 1);
+    TS_ASSERT_DELTA(dupsampletable->cell<double>(0, 2), -0.0002, 0.000001);
+
+    // Clean
+    AnalysisDataService::Instance().remove(outwsname);
+  }
+
+  //-----------------------------------------------------------------------------
+  /**
+   * @brief Test main features required by the algorithm
+   * CorelliPowderCalibrationDatabase
+   */
+  void test_exec() {
+
+    // Create the test environment
+    // create directory
+    std::string calibdir{"TestCorelliPowderCalibration1117"};
+    // clean previous
+    boost::filesystem::remove_all(calibdir);
+    // create data base
+    boost::filesystem::create_directory(calibdir);
+    // create a previously generated database file
+    std::vector<std::string> banks{"source", "sample", "bank2", "bank42"};
```

Comment:
to be consistent, can you use CORELLI's component names for source and sample ("moderator" and "sample-position")?